tools/infer/utility.py

```diff
@@ -232,21 +232,28 @@ def create_predictor(args, mode, logger):
     else:
         file_names = ["model", "inference"]
         for file_name in file_names:
-            model_file_path = "{}/{}.pdmodel".format(model_dir, file_name)
-            params_file_path = "{}/{}.pdiparams".format(model_dir, file_name)
-            if os.path.exists(model_file_path) and os.path.exists(params_file_path):
+            params_file_path = f"{model_dir}/{file_name}.pdiparams"
+            if os.path.exists(params_file_path):
                 break
-        if not os.path.exists(model_file_path):
-            raise ValueError(
-                "not find model.pdmodel or inference.pdmodel in {}".format(model_dir)
-            )
+
         if not os.path.exists(params_file_path):
+            raise ValueError(f"not find {file_name}.pdiparams in {model_dir}")
+
+        if not (
+            os.path.exists(f"{model_dir}/{file_name}.pdmodel")
+            or os.path.exists(f"{model_dir}/{file_name}.json")
+        ):
             raise ValueError(
-                "not find model.pdiparams or inference.pdiparams in {}".format(
-                    model_dir
-                )
+                f"neither {file_name}.json nor {file_name}.pdmodel was found in {model_dir}."
             )
 
+        if not os.path.exists(f"{model_dir}/{file_name}.pdmodel") and os.path.exists(
```

Comment:
```suggestion
        if os.path.exists(
```

no need to recheck whether f"{model_dir}/{file_name}.pdmodel" exists ?